napari/layers/base/base.py

```diff
@@ -721,6 +716,18 @@ def extent(self) -> Extent:
             step=abs(data_to_world.scale),
         )
 
+    def _clear_extent(self):
+        """Clears the cached extent.
+
+        This should be called whenever this data or transform information
+        changes, and should be called before any related events get emitted
+        so that they use the updated extent values.
+        """
+        if 'extent' in self.__dict__:
+            del self.extent
```

Comment:
This is how cached property works. 
The whole mechanism is first to check if the name is defined in `self.__dict__` and if no, then call the property and put the result in this dict. 
None is the proper value, so it will not trigger a recalculation.